### DISCUSSION TITLE
Archway handler integration with Relay

### DISCUSSION THREAD

**Discussion by viveksharmapoudel** [icon-project/IBC-Integration#127] [General]
Hi Team,
I initiated this discussion to discuss about the interaction between relay and archway chain. I would appreciate it if you could kindly provide answers to the following questions:

1. Will the method names of the handler contract be similar to those used in IBC specs?
2. What will be the event names for channel-handshake, connection-handshake, sendPacket, and ack-packet?
3. How can relay obtain the commitment packet proof from archway chain?

**Comment by shreyasbhat0** [comment]
@viveksharmapoudel  to be more specific on the queries you have listed.
As we are developing the entire IBC core as a cosmwasm wasm contract we need to define a message schema for contract execution, contract query and contract instantiate. 
If you give some insights on what all the methods relay will be querying and executing on the contract we can come up with that schema and expose it to the entry point.

**Reply by viveksharmapoudel** [reply]
From the relay perspective, providing method similar to  javascore will suffice... https://github.com/icon-project/IBC-Integration/tree/feature/ICON-mock-lightclient/contracts/javascore/ibc/src/main/java/ibc/ics25/handler ... I am not entirely sure about commitment proofs...since it will be needed by tendermint lightclient, what do you think @AntonAndell ?

**Reply by AntonAndell** [reply]
I am not very well read up on commitment proofs. Mabye @nightowl121 will know more soon.

**Comment by shreyasbhat0** [comment]
2. For events we are following event names specified in the Ibc-rs Library. [Events which we have defined according to that ](https://github.com/icon-project/IBC-Integration/blob/main/contracts/cosmwasm-vm/cw-ibc-core/src/ics04_channel/events.rs)
